web/components/roomConfig/GenericWebhookConfig.tsx

```diff
@@ -83,13 +68,11 @@ const ConnectionConfiguration: FunctionComponent<{
                 }}
             />
             <p> See the <a target="_blank" rel="noopener noreferrer" href={DOCUMENTATION_LINK}>documentation</a> for help writing transformation functions </p>
-        </div>}
-
-        <div className={style.buttonSet}>
-            { canEdit && <Button onClick={onSaveClick}>{ existingConnection ? "Save" : "Add webhook"}</Button>}
+        </InputField>
+        <ButtonSet>
+            { canEdit && <Button onClick={onSaveClick}>{ existingConnection ? "Save" : "Add webhook" }</Button>}
```

Comment:
Because you're not using HTML's `<form>` (which would be awesome for keyboard accessibility, e.g. pressing Ctrl+Enter in an input field submits a form), all your buttons should have `type="button"`. The default is `type="submit"` and they will try to `submit` their parent `<form>`, if there is one.

Beause every input handling is done is JS, you could also add `type="button"` as a default to `Button.tsx`. Many React UI libs do this.

Reply:
I've made this a form instead, and hopefully done it right.